connectors/bigtable/README.md

```diff
@@ -0,0 +1,180 @@
+* `BIGINT`
+* `DOUBLE`
+* `FLOAT`
+* `BYTES`
+* `BINARY`
+* `TIMESTAMP`
+* `TIMESTAMP_WITH_TIME_ZONE`
+* `TIMESTAMP_WITH_LOCAL_TIME_ZONE`
+* `INTERVAL(YEAR, MONTH)` 
+* `INTERVAL(DAY, SECOND)`
+* `TIME`
+* `DATE`
+* `DECIMAL` 
+
+The maximum precision for time-based types is 6.
+
+## Table API
+
+This connector provides support for Flink's Table API, enabling easy and efficient data writing to Bigtable tables within your Flink Table API pipelines. 
+
+The connector is named `bigtable`.
+
+### Row Key
+
+The Bigtable Flink Connector uses the primary key defined in your Flink schema as the row key for writing data to Bigtable. It must be of type `STRING` and cannot be null. Only one primary key can be used.
+
+```
+Schema schema =
+        Schema.newBuilder()
+                .column("my-key", DataTypes.STRING().notNull())
+                .column("stringColumn", DataTypes.STRING())
+                .column("intColumn", DataTypes.INT())
+                .primaryKey("my-key")
+                .build();
+```
+
+#### Connector Options
+
+The following connector options are available:
+
+| Option | Description |
+|---|---|
+| `project` | (Required) The Google Cloud project ID. |
+| `instance` | (Required) The Bigtable instance ID. |
+| `table` | (Required) The Bigtable table ID. |
+| `column-family` | The column family to write to (incompatible with `use-nested-rows-mode`). |
+| `use-nested-rows-mode` | Whether to use nested rows as column families (incompatible with `column-family`). |
+| `sink.parallelism` | The parallelism of the sink. |
+
+## Exactly Once
+
+Due to Bigtable's nature, the Sink offers Exactly Once out of the box. In order to get Exactly Once, every element needs to have a timestamp which will be used in the `RowMutationEntry`.
+
+Bigtable storage format is a tuple of `(Key, CFamily, Column, Timestamp)` and `Value`. Bigtable is idempotent through timestamp. This means that if you write the same value multiple times with the same timestamp, the database doesn't change. If you write different values for the same rowkey, Bigtable will update its state with the value that has the latest timestamp (when querying at now).
+
+| Incoming Mutation (in order)                                  | Bigtable State (after mutation, at ts NOW) |
+| :------------------------------------------------------------ | :---------------------------------------- |
+| RowKey, CF, C, timestamp1, my-value                           | my-value                                   |
+| RowKey, CF, C, timestamp2, my-value-2                         | my-value2                                  |
+| RowKey, CF, C, timestamp1, my-value                           | my-value2                                  |
+| RowKey, CF, C, timestamp3, my-value                           | my-value                                   |
+
+This means that, as long as the timestamp doesn't change with retries (i.e., it is idempotent with retries), Exactly Once works the same as At Least Once.
```

Comment:
we should include an example of how to ensure that the timestamp is applied to the element for exactly-once

Reply:
Talked offline. WE are doing this in a separate PR